packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -138,12 +206,25 @@ export class Trigger {
       const pushFilter = this.props.gitConfiguration.pushFilter;
 
       const push: CfnPipeline.GitPushFilterProperty[] | undefined = pushFilter?.map(filter => {
-        const tags: CfnPipeline.GitTagFilterCriteriaProperty | undefined = {
-          // set to undefined if empty array because CloudFormation does not accept empty array
-          excludes: filter.tagsExcludes?.length ? filter.tagsExcludes : undefined,
-          includes: filter.tagsIncludes?.length ? filter.tagsIncludes : undefined,
-        };
-        return { tags };
+        const tags: CfnPipeline.GitTagFilterCriteriaProperty | undefined =
+          filter.tagsExcludes?.length || filter.tagsIncludes?.length ? {
+            // set to undefined if empty array because CloudFormation does not accept empty array
+            excludes: filter.tagsExcludes?.length ? filter.tagsExcludes : undefined,
+            includes: filter.tagsIncludes?.length ? filter.tagsIncludes : undefined,
+          } : undefined;
```

Comment:
The original code was as follows. It would be better if `tags` were undefined if there were neither `tagsExcludes` nor `tagsIncludes`, but this is not the case. (The `tags` will be an empty object.)

```ts
        const tags: CfnPipeline.GitTagFilterCriteriaProperty | undefined = {
          // set to undefined if empty array because CloudFormation does not accept empty array
          excludes: filter.tagsExcludes?.length ? filter.tagsExcludes : undefined,
          includes: filter.tagsIncludes?.length ? filter.tagsIncludes : undefined,
        };
```

In other words, this change will result in better code. However, the CloudFormation template will be changed in the following usage cases.

```ts
  triggers: [{
    providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
    gitConfiguration: {
      sourceAction: sourceAction,
      // here
      pushFilter: [{
        tagsExcludes: [],
        tagsIncludes: [],
      }],
      // or
      // pushFilter: [{}],
    },
  }],
```

With this CDK code, by the change in this PR, the CFn template would be changed as follows.

```diff
                "Triggers": [
                    {
                        "GitConfiguration": {
                            "Push": [
                                {
-                                   "Tags": {}
                                }
                            ],
                            "SourceActionName": "CodeStarConnectionsSourceAction"
                        },
                        "ProviderType": "CodeStarSourceConnection"
                    }
                ]
```

However, in fact, such a CFn code, i.e., `Tags` is an empty object, will cause the following CFn error and deployments will fail. In other words, this kind of CFn template change cannot happen in a successful stack of existing users. Therefore, there is no need to view this change negatively. (This change case cannot exist in users environments.)

```
The trigger configuration for the source action is not valid. Make sure to choose one trigger configuration for each source action.
```

Reply:
```
  triggers: [{
    providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
```
How do you pass the code star connection? Do you do `providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION("dummy-code-star-connection-arn")` ?

Reply:
Here is the sample code.
https://github.com/aws/aws-cdk/blob/61c7edc66c20066ae5af5b7b09d86a270755f52e/packages/%40aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-with-pipeline-triggers.ts

You can use the `CodeStarConnectionsSourceAction` construct with provider type `CODE_STAR_SOURCE_CONNECTION ` in the `aws-cdk-lib/aws-codepipeline-actions` module.

```ts
import * as cpactions from 'aws-cdk-lib/aws-codepipeline-actions';

const sourceAction2 = new cpactions.CodeStarConnectionsSourceAction({
  actionName: 'CodeStarConnectionsSourceAction2',
  output: sourceOutput2,
  connectionArn,
  owner: 'go-to-k',
  repo: 'cdk-codepipeline-demo-2',
});
```

And you can set the action construct into `sourceAction` in `gitConfiguration`.

```ts
new codepipeline.Pipeline(stack, 'Pipeline2', {
  // ...
  // ...
  triggers: [{
    providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
    gitConfiguration: {
      sourceAction: sourceAction2, // <- here
      pushFilter: [{
        branchesExcludes: ['exclude1', 'exclude2'],
        branchesIncludes: ['include1', 'include2'],
        filePathsExcludes: ['/path/to/exclude1', '/path/to/exclude2'],
        filePathsIncludes: ['/path/to/include1', '/path/to/include2'],
      }],
    },
  }],
});
```